src/igraph_eulerian.c

```diff
@@ -0,0 +1,499 @@
+        is_eulerian_directed(graph, has_path, has_cycle);
+    } else {
+        is_eulerian_undirected(graph, has_path, has_cycle);
+    } 
+    return IGRAPH_SUCCESS;
+}
+
+igraph_bool_t check_if_bridge(igraph_t *g, igraph_integer_t start, igraph_integer_t v, long edge) {
+
+    igraph_vector_t bridges;
+    igraph_integer_t i;
+
+    IGRAPH_CHECK(igraph_vector_init(&bridges, 0));
+    IGRAPH_FINALLY(igraph_vector_destroy, &bridges);
+    igraph_bridges(g, &bridges);
+
+    for (i = 0; i < igraph_vector_size(&bridges); i++) {
+        if (VECTOR(bridges)[i] == edge) {
+            igraph_vector_destroy(&bridges);
+            IGRAPH_FINALLY_CLEAN(1);
+            return 0;   
+        }
+    }
+
+    igraph_vector_destroy(&bridges);
+    IGRAPH_FINALLY_CLEAN(1);
+
+    return 1;
+}
+
+int print_euler_undirected_implementation(igraph_integer_t start, igraph_t *g, igraph_vector_t *path, igraph_vector_bool_t *visited_list) {
+
+    igraph_integer_t i, v, adj;
+    igraph_inclist_t il;
+    igraph_vector_int_t *incedges;
+    long nc, edge;
+    int j;
+
+    i = 0;
+
+    while (1) {
+        IGRAPH_CHECK(igraph_inclist_init(g, &il, IGRAPH_ALL)); 
+        IGRAPH_FINALLY(igraph_inclist_destroy, &il); 
+        incedges = igraph_inclist_get(&il, start);
+        nc = igraph_vector_int_size(incedges);
+        adj = 0;
+
+        for (j = 0; j < nc; j++) {
+            edge = (long) VECTOR(*incedges)[j];
+            if (!VECTOR(*visited_list)[edge]) adj++;
+        }
+
+        if (adj == 0) break;
+
+        edge = (long) VECTOR(*incedges)[i];
+        if (!VECTOR(*visited_list)[edge]) {
+            v = IGRAPH_TO(g, edge) == start ? IGRAPH_FROM(g, edge) : IGRAPH_TO(g, edge);
+            if (adj == 1 || check_if_bridge(g, start, v, edge)) {
+                VECTOR(*visited_list)[edge] = 1;
+                IGRAPH_CHECK(igraph_vector_push_back(path, edge));               
+                print_euler_undirected_implementation(v, g, path, visited_list);
+            }
+        }
+        i++;
+    }
+
+
+    igraph_inclist_destroy(&il);
+    IGRAPH_FINALLY_CLEAN(1);
+
+    return IGRAPH_SUCCESS;
+
+}
+
+
+/* solution adapted from https://www.geeksforgeeks.org/fleurys-algorithm-for-printing-eulerian-path/ */
+int igraph_euler_path_undirected(igraph_t *graph, igraph_vector_t *path) {
+
+    /* default starting vertex is 0
+    when we have no odd vertices, we can start anywhere (usually 0)
+    when we have 2 odd vertices, we start at any one of them
+    igraph_vector_t path must be initialised prior to its us */
+
+    igraph_integer_t start, i, e_count;
+    igraph_t copy; 
+    igraph_inclist_t incl;
+    igraph_vector_int_t *incedges;
+    igraph_bool_t has_path;
+    igraph_bool_t cycle;
+    igraph_vector_bool_t visited_list;
+
+    has_path = 0;
+    cycle = 0;
+
+    igraph_is_eulerian(graph, &has_path, &cycle);
+
+    if (!has_path && !cycle) {
+        IGRAPH_WARNING("Euler cycle not possible");
+        return IGRAPH_FAILURE;
+    }
```

Comment:
Is this necessary? It seems to me that all public functions that call this already do the check. Did I miss anything?

Reply:
My bad, forgot to remove this.